### PR TITLE
Build using Docker

### PR DESCRIPTION
**Pull request by ehazlett**
This updates the build process to build using containers.  This removes the potential for host level conflicts and makes it repeatable for local development.

### CODE REVIEW

File: .github/workflows/publish-prod.yml [hunk 1]
```diff
@@ -64,20 +54,17 @@ jobs:
...
-
-      - name: Publish docker image
-        run: |
-          docker tag ktranslate:v2 kentik/ktranslate:v2
-          docker push kentik/ktranslate:v2
-          docker tag ktranslate:v2 kentik/ktranslate:${{ env.KENTIK_KTRANSLATE_VERSION }}
-          docker push kentik/ktranslate:${{ env.KENTIK_KTRANSLATE_VERSION }}
-          docker tag ktranslate:v2 kentik/ktranslate:latest
-          docker push kentik/ktranslate:latest
+      - name: Build and Publish Docker
```
Comment (ehazlett) [inline .github/workflows/publish-prod.yml:57]: @i3149 If this is OK, we will need to add a `build-arg` to use the `MAXMIND_LICENSE_KEY` env var in the Dockerfile.

Comment (i3149) [reply .github/workflows/publish-prod.yml:57]: Makes sense to me.

File: Dockerfile [hunk 2]
```diff
@@ -1,26 +1,57 @@
...
+WORKDIR /src
+RUN make
+
+# maxmind dbs
+FROM alpine:latest as maxmind
+ARG MAXMIND_LICENSE_KEY
+RUN apk add -U curl tar
+ENV GEOLITE2_COUNTRY_FILE=GeoLite2-Country.mmdb
+ENV GEOLITE2_ASN_FILE=GeoLite2-ASN.mmdb
+RUN if [ -z "${MAXMIND_LICENSE_KEY}" ]; then echo "MAXMIND_LICENSE_KEY" not set; exit 1; fi
```
Comment (i3149) [inline Dockerfile:14]: Makes sense to pull it down here.

File: Dockerfile [hunk 3]
```diff
@@ -1,26 +1,57 @@
...
 	useradd --home-dir /etc/ktranslate --gid ktranslate --no-create-home --uid 1000 ktranslate
 
 COPY --chown=ktranslate:ktranslate config/ /etc/ktranslate/
+COPY --chown=ktranslate:ktranslate lib/ /etc/ktranslate/
+
+# maxmind db
+COPY --from=maxmind /GeoLite2-Country.mmdb /etc/ktranslate/
+COPY --from=maxmind /GeoLite2-ASN.mmdb /etc/ktranslate/
+# snmp
+COPY --from=snmp /snmp/profiles /etc/ktranslate/profiles/profiles
```
Comment (i3149) [inline Dockerfile:45]: To clean this up a bit, possibly we should make the path just `/etc/ktranslate/profiles` without the double profiles? This was a bug I'd been meaning to get around to.

File: DockerfileArm64 [hunk 4]
```diff
@@ -1,29 +0,0 @@
...
-	useradd --home-dir /etc/ktranslate --gid ktranslate --no-create-home --uid 1000 ktranslate
-
-COPY --chown=ktranslate:ktranslate config/ /etc/ktranslate/
-
-# add backwards compatibility symlinks for folks using an snmp.yml from the older image (and "ls" to verify the symlinks are correct and working)
-RUN set -eux; ln -sv ktranslate/profiles ktranslate/mibs.db /etc/; ls -l /etc/profiles/ /etc/mibs.db/
-RUN set -eux; ln -sv /etc/ktranslate/mibs.db /etc/mib.db; ls -l /etc/mib.db/
-
-# Fix libpcap to work with how we built things.
-RUN set -eux; ln -sv /usr/lib/aarch64-linux-gnu/libpcap.so.1.9.1 /usr/lib/aarch64-linux-gnu/libpcap.so.1; ls -l /usr/lib/aarch64-linux-gnu/libpcap.so.1
```
Comment (i3149) [inline DockerfileArm64:22]: Would we still need this line?

File: Dockerfile [hunk 5]
```diff
@@ -1,26 +1,57 @@
...
+RUN curl -o /tmp/asn.tar.gz "https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-ASN&license_key=${MAXMIND_LICENSE_KEY}&suffix=tar.gz" && \
+	tar zxf /tmp/asn.tar.gz --strip-components 1 -C /
+
+# snmp profiles
+FROM alpine:latest as snmp
+RUN apk add -U git
+RUN git clone https://github.com/kentik/snmp-profiles /snmp
 
+# main image
+FROM ubuntu:20.04
```
Comment (i3149) [inline Dockerfile:26]: Unrelated, but I've been thinking about moving to alpine from ubuntu to reduce the vulnerability scan count. Thoughts?

**Review by i3149 — APPROVED** [review]
LGTM -- thanks for taking this on!

File: DockerfileArm64 (hunk 4 above)
Comment (ehazlett) [reply DockerfileArm64:22]: In my testing on my m1 the image built and ran fine using the multi-arch ubuntu.

File: Dockerfile (hunk 3 above)
Comment (ehazlett) [reply Dockerfile:45]: +1 I noticed that.  I'll remove the extra dir :)

File: Dockerfile (hunk 5 above)
Comment (ehazlett) [reply Dockerfile:26]: I'm always happy to move to Alpine :). I'll switch and see if there are any issues.

File: DockerfileArm64 (hunk 4 above)
Comment (i3149) [reply DockerfileArm64:22]: awsome.

**Review by i3149 — APPROVED** [review]
:rocket:

**Pull request merged**